Assets/MixedRealityToolkit/_Core/Utilities/Lines/Renderers/MeshLineRenderer.cs

```diff
@@ -0,0 +1,233 @@
+
+        public Mesh LineMesh
+        {
+            get { return lineMesh; }
+            set
+            {
+                enabled = false;
+                lineMesh = value;
+                enabled = true;
+            }
+        }
+
+        [SerializeField]
+        private Material lineMaterial = null;
+
+        public Material LineMaterial
+        {
+            get { return lineMaterial; }
+            set
+            {
+                enabled = false;
+                lineMaterial = value;
+                enabled = true;
+            }
+        }
+
+        [SerializeField]
+        private string colorProp = "_Color";
+
+        public string ColorProp
+        {
+            get { return colorProp; }
+            set
+            {
+                enabled = false;
+                colorProp = value;
+
+                if (!lineMaterial.HasProperty(value))
+                {
+                    Debug.LogError($"Unable to find the property {value} for the line material");
+                    return;
+                }
+
+                enabled = true;
+            }
+        }
+
+        #region Command buffer properties
+
+        private readonly Dictionary<Camera, CommandBuffer> cameras = new Dictionary<Camera, CommandBuffer>();
+
+        private int colorId;
+        private Vector4[] colorValues;
+        private Matrix4x4[] meshTransforms;
+        private bool executeCommandBuffer = false;
+        private MaterialPropertyBlock linePropertyBlock;
+
+        #endregion
+
+        #region OnWillRenderObject helpers
+
+        private readonly Vector3[] meshVertices = new Vector3[3];
+
+        [SerializeField]
+        [HideInInspector]
+        private Material renderedMaterial = null;
+
+        [SerializeField]
+        [HideInInspector]
+        private Mesh renderedMesh = null;
+
+        private MeshRenderer meshRenderer;
+
+        #endregion OnWillRenderObject helpers
+
+        private void OnValidate()
+        {
+            if (lineMaterial == null)
+            {
+                Debug.LogError("LineDataProvider material cannot be null.");
+            }
+
+            if (lineMesh == null)
+            {
+                Debug.LogError("LineDataProvider mesh cannot be null.");
+            }
+
+            if (renderedMaterial == null)
+            {
+                // Create an 'invisible' material so the mesh doesn't show up pink
+                renderedMaterial = new Material(Shader.Find(InvisibleShaderName));
+            }
+
+            if (renderedMesh == null)
+            {
+                // create a simple 1-triangle mesh to ensure OnWillRenderObject is always called.
+                renderedMesh = new Mesh
+                {
+                    vertices = meshVertices,
+                    triangles = new[] { 0, 1, 2 }
+                };
+            }
+        }
+
+        protected void OnEnable()
+        {
+            if (lineMaterial == null)
```

Comment:
Can we genericise this base call in to something like
```AssureDependency(object value, string message) { if (object == null) { Debug.Log(message);} ```
To reduce the sheer amount of duplicated calls to Nullcheck /  Debug.Log in this script.

Reply:
I don't understand.  Besides this check only happens on enable, so it doesn't have that much overhead.

Reply:
It's just that we have "if null / Debug" repeated everywhere.

Reply:
I added a bool that checks the values for you.